src/executor.ts

```diff
@@ -2,19 +2,23 @@ import chalk from "chalk";
 import {Job} from "./job";
 import assert, {AssertionError} from "assert";
 import {Argv} from "./argv";
+import {PromisePool} from "@supercharge/promise-pool";
 
 export class Executor {
 
     static async runLoop (argv: Argv, jobs: ReadonlyArray<Job>, stages: readonly string[], potentialStarters: Job[]) {
         let startCandidates = [];
 
         do {
-            const promises = [];
             startCandidates = Executor.getStartCandidates(jobs, stages, potentialStarters, argv.manual);
-            for (const s of startCandidates) {
-                promises.push(s.start());
+            if (startCandidates.length > 0) {
+                await PromisePool
+                    .withConcurrency(argv.concurrency ?? startCandidates.length)
```

Comment:
suggestion: use `os.cpus().length - 1` as the default value? Pass `-1` for unlimited concurrency?

Reply:
If developers have problems with concurrency they can explicitly set it.

People have been running with unlimited up until now, no need to change that default.